### PR TITLE
Switch to Auth to get GitHub tokens

### PR DESCRIPTION
**Pull request by alexeykazakov**
Related to https://github.com/fabric8io/fabric8-generator/issues/70

### CODE REVIEW

**Comment by alexeykazakov** [comment]
@jstrachan please take a look

File: src/main/java/io/fabric8/forge/generator/keycloak/KeycloakEndpoint.java [hunk 1]
```diff
@@ -26,15 +27,15 @@
     // http://sso.prod-preview.openshift.io/auth/realms/fabric8/account
     // authorization: Bearer <ACCESS_TOKEN>
     // http://sso.prod-preview.openshift.io/auth/realms/fabric8/account/identity
-    GET_OPENSHIFT_TOKEN("OpenShift", "/auth/realms/fabric8/broker/openshift-v3/token"),
-    GET_GITHUB_TOKEN("GitHub", "/auth/realms/fabric8/broker/github/token");
+    GET_OPENSHIFT_TOKEN("OpenShift", URLUtils.pathJoin(Configuration.keycloakSaaSURL(), "/auth/realms/fabric8/broker/openshift-v3/token")),
```
Comment (sbose78) [inline src/main/java/io/fabric8/forge/generator/keycloak/KeycloakEndpoint.java:30]: I have a question: 
Previously we only had `"/auth/realms/fabric8/broker/openshift-v3/token"`
and now we have `URLUtils.pathJoin(Configuration.keycloakSaaSURL(), "/auth/realms/fabric8/broker/openshift-v3/token"))` .

How did it work before with only the path? :)

Comment (alexeykazakov) [reply src/main/java/io/fabric8/forge/generator/keycloak/KeycloakEndpoint.java:30]: `GET_OPENSHIFT_TOKEN()` calls the enumeration constructor `KeycloakEndpoint(String name, String path)` where we had `this.endpoint = URLUtils.pathJoin(Configuration.keycloakSaaSURL(), path);`
Now we are removing it from constructor. See lines 37-38.
It's a Java thing ;)

Comment (sbose78) [reply src/main/java/io/fabric8/forge/generator/keycloak/KeycloakEndpoint.java:30]: Oh right :D my mind was seeing this change in reverse - thanks for pointing out.
```
-        this.endpoint = URLUtils.pathJoin(Configuration.keycloakSaaSURL(), path);
+        this.endpoint = url;
```

**Comment by jstrachan** [comment]
LGTM

**Pull request merged**